### ISSUE TITLE
.2 Fails to build on MacOS

### ISSUE THREAD

**Issue by edgework** [jduncanator/node-diskusage#16]
.2 version fails to build on MaxOS.  .1.5 builds fine.   This with with latest version of XCode Command Line tools on 10.12.2

  CXX(target) Release/obj.target/diskusage/src/main.o
In file included from ../src/main.cpp:1:
../src/diskusage.h:3:10: fatal error: 'cstdint' file not found
#include <cstdint>
         ^
1 error generated.
make: *** [Release/obj.target/diskusage/src/main.o] Error 1`

**Comment by miniak** [comment]
That's weird. I tested it on latest macOS with latest Xcode installed.

**Comment by edgework** [comment]
Maybe its node version?  I'm on  v4.7.0

**Comment by miniak** [comment]
I will have a look tomorrow, it's already too late for me in Prague. I can refactor the code to avoid using that header.

**Comment by miniak** [comment]
Or it could be one of these issues:
http://stackoverflow.com/questions/10116724/clang-os-x-lion-cannot-find-cstdint
http://stackoverflow.com/questions/16525489/cstdint-file-not-found-when-building-on-the-command-line

**Comment by miniak** [comment]
You can try cloning the git repo and calling npm install. Try changing the include to stdint.h. If that helps, you can submit a PR (bump the version to 0.2.1)

**Comment by jduncanator** [comment]
I tested it with nvm on macOS Sierra with an old-ish XCode (probably 6 months old) before publishing the changes and everything was good from Node v0.10.48 all the way up to latest stable.

I'll update my XCode version and see how I go.

@miniak There was an issue with the cflags on older Node versions (node-gyp didn't invoke `-std=c++11` by default in those versions) so this may be affecting MacOS. I think I remember there being an issue somewhere that gyp on Mac ignored the cflags option in the gyp file and you had to do a work around. May be related, will check it out.

**Comment by jduncanator** [comment]
Was able to reproduce on v0.10.48 with El Capitan and its latest XCode. Definitely related to missing C++ cflag. Will make some changes.

**Comment by jduncanator** [comment]
Turns out its not related to cflags.

@miniak It seems like new versions of node are fine with building it out of the box, but older node versions require the include in `diskusage.h` to be `#include <tr1/cstdint>` instead of `#include <cstdint>`. Still not sure why. Do you have any ideas?

If you want to reproduce, just install nvm and grab 0.10.48 (`nvm install 0.10.48`) and try and do a `npm install` inside the git repository.

**Comment by edgework** [comment]
Hey, do you have any idea when you'll push an update to NPM?

**Comment by jduncanator** [comment]
@edgework When we determine what the issue is, we'll push an update. Currently, updating to the latest version of Node will solve the problem, so if you can't wait, I suggest updating.

**Comment by edgework** [comment]
There's no way we can update Node. We're on 4.4 which is the most common production version out there.   I just locked the version down on all our servers to 0.1.x

**Comment by jduncanator** [comment]
@edgework We're running v6 in production, which should install diskusage fine. If you are deploying to production, you should have shrink-wrapped dependencies, or at least a fixed version in your package.json.

I'm working on resolving the issue now, but without @miniak's input, it's going to take a while.

**Comment by miniak** [comment]
Did you try just replacing the include with stdint.h?

**Comment by jduncanator** [comment]
@miniak From my testing, doesn't work when compiling with C++11 (which is the default under new node's).

**Comment by miniak** [comment]
So which node versions should I test with? 4 and 6?

**Comment by jduncanator** [comment]
@miniak Test with v0.10.48, v4 and v7. That should cover everything from what I can tell.

**Comment by jduncanator** [comment]
>If you want to reproduce, just install nvm and grab 0.10.48 (nvm install 0.10.48) and try and do a npm install inside the git repository.

**Comment by miniak** [comment]
I was able to reproduce with v4

  CXX(target) Release/obj.target/diskusage/src/main.o
In file included from ../src/main.cpp:1:
../src/diskusage.h:3:10: fatal error: 'ctdint' file not found
#include <ctdint>

**Comment by miniak** [comment]
For it works using `stdint.h` with both v4 and v7

**Comment by jduncanator** [comment]
@miniak Doesn't work on Linux using `stdint.h` on node v7

**Comment by miniak** [comment]
I only have access to macOS and Windows at this point.

**Comment by miniak** [comment]
What about using `#ifdef` for that include?

**Comment by miniak** [comment]
>> Doesn't work on Linux using stdint.h on node v7

can you send me the error message that you get?

**Comment by jduncanator** [comment]
@miniak Can't reproduce now. Not sure what I was doing wrong last time but `<stdint.h>` works fine for Linux. Happy to make that change, and see what happens.

**Comment by miniak** [comment]
The other option would be to stop using `cstdint` / `stdint.h` altogether and change

```
struct DiskUsage {
    uint64_t available;
    uint64_t free;
    uint64_t total;
};
```

to

```
struct DiskUsage {
    double available;
    double free;
    double total;
};
```

It needs to be passed as double to JavaScript anyway.

**Comment by jduncanator** [comment]
Will see if this change works for @edgework and will then look at changing to double.

**Comment by jduncanator** [comment]
@miniak Get some interesting output from Linux with Node v0.12.18

```
make: Entering directory '/root/node-diskusage/build'
  CXX(target) Release/obj.target/diskusage/src/main.o
In file included from /root/.node-gyp/0.12.18/include/node/v8stdint.h:13:0,
                 from /root/.node-gyp/0.12.18/include/node/v8.h:18,
                 from /root/.node-gyp/0.12.18/include/node/node.h:61,
                 from ../node_modules/nan/nan.h:47,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node.h: In function ‘void node::FatalException(const v8::TryCatch&)’:
/root/.node-gyp/0.12.18/include/node/node.h:275:61: warning: ‘void node::FatalException(v8::Isolate*, const v8::TryCatch& ’ is deprecated: Use ParseEncoding(isolate, ...) [-Wdeprecated-declarations]
   return FatalException(v8::Isolate::GetCurrent(), try_catch);
                                                             ^
/root/.node-gyp/0.12.18/include/node/v8config.h:313:1: note: in definition of macro ‘V8_DEPRECATED’
 declarator __attribute__((deprecated(message)))
 ^
/root/.node-gyp/0.12.18/include/node/node.h:273:1: note: in expansion of macro ‘NODE_DEPRECATED’
 NODE_DEPRECATED("Use FatalException(isolate, ...)",
 ^
In file included from ../node_modules/nan/nan.h:47:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node.h:270:18: note: declared here
 NODE_EXTERN void FatalException(v8::Isolate* isolate,
                  ^
In file included from /root/.node-gyp/0.12.18/include/node/v8stdint.h:13:0,
                 from /root/.node-gyp/0.12.18/include/node/v8.h:18,
                 from /root/.node-gyp/0.12.18/include/node/node.h:61,
                 from ../node_modules/nan/nan.h:47,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node.h: In function ‘v8::Local<v8::Value> node::Encode(const void*, size_t, node::encoding)’:
/root/.node-gyp/0.12.18/include/node/node.h:287:62: warning: ‘v8::Local<v8::Value> node::Encode(v8::Isolate*, const void*, size_t, node::encoding)’ is deprecated: Use FatalException(isolate, ...) [-Wdeprecated-declarations]
   return Encode(v8::Isolate::GetCurrent(), buf, len, encoding);
                                                              ^
/root/.node-gyp/0.12.18/include/node/v8config.h:313:1: note: in definition of macro ‘V8_DEPRECATED’
 declarator __attribute__((deprecated(message)))
 ^
/root/.node-gyp/0.12.18/include/node/node.h:282:1: note: in expansion of macro ‘NODE_DEPRECATED’
 NODE_DEPRECATED("Use Encode(isolate, ...)",
 ^
In file included from ../node_modules/nan/nan.h:47:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node.h:278:34: note: declared here
 NODE_EXTERN v8::Local<v8::Value> Encode(v8::Isolate* isolate,
                                  ^
In file included from /root/.node-gyp/0.12.18/include/node/v8stdint.h:13:0,
                 from /root/.node-gyp/0.12.18/include/node/v8.h:18,
                 from /root/.node-gyp/0.12.18/include/node/node.h:61,
                 from ../node_modules/nan/nan.h:47,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node.h: In function ‘ssize_t node::DecodeBytes(v8::Handle<v8::Value>, node::encoding ’:
/root/.node-gyp/0.12.18/include/node/node.h:298:62: warning: ‘ssize_t node::DecodeBytes(v8::Isolate*, v8::Handle<v8::Value>, node::encoding)’ is deprecated: Use Encode(isolate, ...) [-Wdeprecated-declarations]
   return DecodeBytes(v8::Isolate::GetCurrent(), val, encoding);
                                                              ^
/root/.node-gyp/0.12.18/include/node/v8config.h:313:1: note: in definition of macro ‘V8_DEPRECATED’
 declarator __attribute__((deprecated(message)))
 ^
/root/.node-gyp/0.12.18/include/node/node.h:294:1: note: in expansion of macro ‘NODE_DEPRECATED’
 NODE_DEPRECATED("Use DecodeBytes(isolate, ...)",
 ^
In file included from ../node_modules/nan/nan.h:47:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node.h:291:21: note: declared here
 NODE_EXTERN ssize_t DecodeBytes(v8::Isolate* isolate,
                     ^
In file included from /root/.node-gyp/0.12.18/include/node/v8stdint.h:13:0,
                 from /root/.node-gyp/0.12.18/include/node/v8.h:18,
                 from /root/.node-gyp/0.12.18/include/node/node.h:61,
                 from ../node_modules/nan/nan.h:47,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node.h: In function ‘ssize_t node::DecodeWrite(char*, size_t, v8::Handle<v8::Value>, node::encoding)’:
/root/.node-gyp/0.12.18/include/node/node.h:312:75: warning: ‘ssize_t node::DecodeWrite(v8::Isolate*, char*, size_t, v8::Handle<v8::Value>, node::encoding)’ is deprecated: Use DecodeBytes(isolate, ...) [-Wdeprecated-declarations]
   return DecodeWrite(v8::Isolate::GetCurrent(), buf, buflen, val, encoding);
                                                                           ^
/root/.node-gyp/0.12.18/include/node/v8config.h:313:1: note: in definition of macro ‘V8_DEPRECATED’
 declarator __attribute__((deprecated(message)))
 ^
/root/.node-gyp/0.12.18/include/node/node.h:307:1: note: in expansion of macro ‘NODE_DEPRECATED’
 NODE_DEPRECATED("Use DecodeWrite(isolate, ...)",
 ^
In file included from ../node_modules/nan/nan.h:47:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node.h:302:21: note: declared here
 NODE_EXTERN ssize_t DecodeWrite(v8::Isolate* isolate,
                     ^
In file included from /root/.node-gyp/0.12.18/include/node/v8stdint.h:13:0,
                 from /root/.node-gyp/0.12.18/include/node/v8.h:18,
                 from /root/.node-gyp/0.12.18/include/node/node.h:61,
                 from ../node_modules/nan/nan.h:47,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node_buffer.h: In function ‘v8::Local<v8::Object> node::Buffer::New(v8::Handle<v8::String>, node::encoding)’:
/root/.node-gyp/0.12.18/include/node/node_buffer.h:58:52: warning: ‘v8::Local<v8::Object> node::Buffer::New(v8::Isolate*, v8::Handle<v8::String>, node::encoding)’ is deprecated: Use New(isolate, ...) [-Wdeprecated-declarations]
   return New(v8::Isolate::GetCurrent(), string, enc);
                                                    ^
/root/.node-gyp/0.12.18/include/node/v8config.h:313:1: note: in definition of macro ‘V8_DEPRECATED’
 declarator __attribute__((deprecated(message)))
 ^
/root/.node-gyp/0.12.18/include/node/node_buffer.h:55:1: note: in expansion of macro ‘NODE_DEPRECATED’
 NODE_DEPRECATED("Use New(isolate, ...)",
 ^
In file included from ../node_modules/nan/nan.h:48:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node_buffer.h:52:35: note: declared here
 NODE_EXTERN v8::Local<v8::Object> New(v8::Isolate* isolate,
                                   ^
In file included from /root/.node-gyp/0.12.18/include/node/v8stdint.h:13:0,
                 from /root/.node-gyp/0.12.18/include/node/v8.h:18,
                 from /root/.node-gyp/0.12.18/include/node/node.h:61,
                 from ../node_modules/nan/nan.h:47,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node_buffer.h: In function ‘v8::Local<v8::Object> node::Buffer::New(const char*, size_t)’:
/root/.node-gyp/0.12.18/include/node/node_buffer.h:67:50: warning: ‘v8::Local<v8::Object> node::Buffer::New(v8::Isolate*, const char*, size_t)’ is deprecated: Use New(isolate, ...) [-Wdeprecated-declarations]
   return New(v8::Isolate::GetCurrent(), data, len);
                                                  ^
/root/.node-gyp/0.12.18/include/node/v8config.h:313:1: note: in definition of macro ‘V8_DEPRECATED’
 declarator __attribute__((deprecated(message)))
 ^
/root/.node-gyp/0.12.18/include/node/node_buffer.h:65:1: note: in expansion of macro ‘NODE_DEPRECATED’
 NODE_DEPRECATED("Use New(isolate, ...)",
 ^
In file included from ../node_modules/nan/nan.h:48:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node_buffer.h:62:35: note: declared here
 NODE_EXTERN v8::Local<v8::Object> New(v8::Isolate* isolate,
                                   ^
In file included from /root/.node-gyp/0.12.18/include/node/v8stdint.h:13:0,
                 from /root/.node-gyp/0.12.18/include/node/v8.h:18,
                 from /root/.node-gyp/0.12.18/include/node/node.h:61,
                 from ../node_modules/nan/nan.h:47,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node_buffer.h: In function ‘v8::Local<v8::Object> node::Buffer::New(char*, size_t, node::smalloc::FreeCallback, void*)’:
/root/.node-gyp/0.12.18/include/node/node_buffer.h:80:69: warning: ‘v8::Local<v8::Object> node::Buffer::New(v8::Isolate*, char*, size_t, node::smalloc::FreeCallback, void*)’ is deprecated: Use New(isolate, ...) [-Wdeprecated-declarations]
   return New(v8::Isolate::GetCurrent(), data, length, callback, hint);
                                                                     ^
/root/.node-gyp/0.12.18/include/node/v8config.h:313:1: note: in definition of macro ‘V8_DEPRECATED’
 declarator __attribute__((deprecated(message)))
 ^
/root/.node-gyp/0.12.18/include/node/node_buffer.h:75:1: note: in expansion of macro ‘NODE_DEPRECATED’
 NODE_DEPRECATED("Use New(isolate, ...)",
 ^
In file included from ../node_modules/nan/nan.h:48:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node_buffer.h:70:35: note: declared here
 NODE_EXTERN v8::Local<v8::Object> New(v8::Isolate* isolate,
                                   ^
In file included from /root/.node-gyp/0.12.18/include/node/v8stdint.h:13:0,
                 from /root/.node-gyp/0.12.18/include/node/v8.h:18,
                 from /root/.node-gyp/0.12.18/include/node/node.h:61,
                 from ../node_modules/nan/nan.h:47,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node_buffer.h: In function ‘v8::Local<v8::Object> node::Buffer::Use(char*, uint32_t)’:
/root/.node-gyp/0.12.18/include/node/node_buffer.h:90:50: warning: ‘v8::Local<v8::Object> node::Buffer::Use(v8::Isolate*, char*, uint32_t)’ is deprecated: Use New(isolate, ...) [-Wdeprecated-declarations]
   return Use(v8::Isolate::GetCurrent(), data, len);
                                                  ^
/root/.node-gyp/0.12.18/include/node/v8config.h:313:1: note: in definition of macro ‘V8_DEPRECATED’
 declarator __attribute__((deprecated(message)))
 ^
/root/.node-gyp/0.12.18/include/node/node_buffer.h:88:1: note: in expansion of macro ‘NODE_DEPRECATED’
 NODE_DEPRECATED("Use Use(isolate, ...)",
 ^
In file included from ../node_modules/nan/nan.h:48:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node_buffer.h:85:35: note: declared here
 NODE_EXTERN v8::Local<v8::Object> Use(v8::Isolate* isolate,
                                   ^
In file included from ../src/main.cpp:3:0:
../node_modules/nan/nan.h: In function ‘Nan::MaybeLocal<v8::Object> Nan::NewBuffer(char*, size_t, node::smalloc::FreeCallback, void*)’:
../node_modules/nan/nan.h:692:34: warning: ‘v8::Local<v8::Object> node::Buffer::New(v8::Isolate*, char*, size_t, node::smalloc::FreeCallback, void*)’ is deprecated: Use New(isolate, ...) [-Wdeprecated-declarations]
                              hint);
                                  ^
In file included from ../node_modules/nan/nan.h:48:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node_buffer.h:70:35: note: declared here
 NODE_EXTERN v8::Local<v8::Object> New(v8::Isolate* isolate,
                                   ^
In file included from ../src/main.cpp:3:0:
../node_modules/nan/nan.h: In function ‘Nan::MaybeLocal<v8::Object> Nan::CopyBuffer(const char*, uint32_t)’:
../node_modules/nan/nan.h:707:67: warning: ‘v8::Local<v8::Object> node::Buffer::New(v8::Isolate*, const char*, size_t)’ is deprecated: Use New(isolate, ...) [-Wdeprecated-declarations]
     return node::Buffer::New(v8::Isolate::GetCurrent(), data, size);
                                                                   ^
In file included from ../node_modules/nan/nan.h:48:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node_buffer.h:62:35: note: declared here
 NODE_EXTERN v8::Local<v8::Object> New(v8::Isolate* isolate,
                                   ^
In file included from ../src/main.cpp:3:0:
../node_modules/nan/nan.h: In function ‘Nan::MaybeLocal<v8::Object> Nan::NewBuffer(char*, uint32_t)’:
../node_modules/nan/nan.h:733:67: warning: ‘v8::Local<v8::Object> node::Buffer::Use(v8::Isolate*, char*, uint32_t)’ is deprecated: Use New(isolate, ...) [-Wdeprecated-declarations]
     return node::Buffer::Use(v8::Isolate::GetCurrent(), data, size);
                                                                   ^
In file included from ../node_modules/nan/nan.h:48:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node_buffer.h:85:35: note: declared here
 NODE_EXTERN v8::Local<v8::Object> Use(v8::Isolate* isolate,
                                   ^
In file included from ../src/main.cpp:3:0:
../node_modules/nan/nan.h: In function ‘void Nan::FatalException(const Nan::TryCatch&)’:
../node_modules/nan/nan.h:868:73: warning: ‘void node::FatalException(v8::Isolate*, const v8::TryCatch&)’ is deprecated: Use ParseEncoding(isolate, ...) [-Wdeprecated-declarations]
     node::FatalException(v8::Isolate::GetCurrent(), try_catch.try_catch_);
                                                                         ^
In file included from ../node_modules/nan/nan.h:47:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node.h:270:18: note: declared here
 NODE_EXTERN void FatalException(v8::Isolate* isolate,
                  ^
In file included from ../src/main.cpp:3:0:
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Value> Nan::Encode(const void*, size_t, Nan::Encoding)’:
../node_modules/nan/nan.h:1786:44: warning: ‘v8::Local<v8::Value> node::Encode(v8::Isolate*, const void*, size_t, node::encoding)’ is deprecated: Use FatalException(isolate, ...) [-Wdeprecated-declarations]
     , static_cast<node::encoding>(encoding));
                                            ^
In file included from ../node_modules/nan/nan.h:47:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node.h:278:34: note: declared here
 NODE_EXTERN v8::Local<v8::Value> Encode(v8::Isolate* isolate,
                                  ^
In file included from ../src/main.cpp:3:0:
../node_modules/nan/nan.h: In function ‘ssize_t Nan::DecodeBytes(v8::Local<v8::Value>, Nan::Encoding)’:
../node_modules/nan/nan.h:1802:44: warning: ‘ssize_t node::DecodeBytes(v8::Isolate*, v8::Handle<v8::Value>, node::encoding)’ is deprecated: Use Encode(isolate, ...) [-Wdeprecated-declarations]
     , static_cast<node::encoding>(encoding));
                                            ^
In file included from ../node_modules/nan/nan.h:47:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node.h:291:21: note: declared here
 NODE_EXTERN ssize_t DecodeBytes(v8::Isolate* isolate,
                     ^
In file included from ../src/main.cpp:3:0:
../node_modules/nan/nan.h: In function ‘ssize_t Nan::DecodeWrite(char*, size_t, v8::Local<v8::Value>, Nan::Encoding)’:
../node_modules/nan/nan.h:1824:44: warning: ‘ssize_t node::DecodeWrite(v8::Isolate*, char*, size_t, v8::Handle<v8::Value>, node::encoding)’ is deprecated: Use DecodeBytes(isolate, ...) [-Wdeprecated-declarations]
     , static_cast<node::encoding>(encoding));
                                            ^
In file included from ../node_modules/nan/nan.h:47:0,
                 from ../src/main.cpp:3:
/root/.node-gyp/0.12.18/include/node/node.h:302:21: note: declared here
 NODE_EXTERN ssize_t DecodeWrite(v8::Isolate* isolate,
                     ^
  CXX(target) Release/obj.target/diskusage/src/diskusage_posix.o
  SOLINK_MODULE(target) Release/obj.target/diskusage.node
  COPY Release/diskusage.node
make: Leaving directory '/root/node-diskusage/build'
nan@2.5.1 node_modules/nan
```

Build still completes, but a lot of deprecation warnings.

**Comment by jduncanator** [comment]
@edgework The latest commit to master has resolved the issue on MacOS with Node v4.4.7 for me. Can you please confirm that you no longer have issues compiling the module?

**Comment by jduncanator** [comment]
@edgework I've published v0.2.1 to NPM. Please try this version and let me know if you run into compilation issues.

**Comment by edgework** [comment]
`$ npm install diskusage
\
> diskusage@0.2.1 install /Users/edgework/node_modules/diskusage
> node-gyp rebuild

  CXX(target) Release/obj.target/diskusage/src/main.o
  CXX(target) Release/obj.target/diskusage/src/diskusage_posix.o
  SOLINK_MODULE(target) Release/diskusage.node
clang: warning: libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9
diskusage@0.2.1 node_modules/diskusage
└── nan@2.5.1
`

Looks good!  Thanks.
